include/libtorrent/alert_types.hpp

```diff
@@ -2066,6 +2067,11 @@ namespace libtorrent
 		boost::array<char, 64> signature;
 		std::string salt;
 		boost::uint64_t seq;
+
+		// DHT put operation usually writes item to k nodes, maybe the node
+		// is stale so no response, or the node doesn't support 'put', or the
+		// token for write is out of date, etc.
```

Comment:
I'll fix the comment.

No, current logic is, get k nodes, do put (same as announce). Even all failed, we won't try more nodes.